config/http_config.go

```diff
@@ -858,117 +854,120 @@ func (rt *basicAuthRoundTripper) CloseIdleConnections() {
 }
 
 type oauth2RoundTripper struct {
+	mtx        sync.RWMutex
+	next       *oauth2.Transport
+	lastSecret string
+
+	// Required for interaction with Oauth2 server.
 	config       *OAuth2
-	rt           http.RoundTripper
-	next         http.RoundTripper
 	clientSecret secret
-	lastSecret   string
-	mtx          sync.RWMutex
 	opts         *httpClientOptions
 	client       *http.Client
 }
 
 func NewOAuth2RoundTripper(clientSecret secret, config *OAuth2, next http.RoundTripper, opts *httpClientOptions) http.RoundTripper {
+	if clientSecret == nil {
+		// TODO(bwplotka): Consider removing support for Oauth2 with just clientID and no secret,
```

Comment:
Note: This is allowed in the OAuth 2 specification although rare. I would remove this comment.

Reply:
Right, I wanted to challenge this, but I can see we do it in other places too, let's kill. I wonder if there is any abstraction which would unify all those roundtrippers to ensure some consistency across them, will check later.